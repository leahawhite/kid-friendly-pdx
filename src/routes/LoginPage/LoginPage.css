.LoginPage {
  text-align: left;
  font-size: 18px;
  font-weight: bold;
  margin: 0 auto;
  padding: 10vh 5vh;
}

.LoginPage h2 {
  text-align: center;
  font-size: 24px;
}

.LoginPage-container {
  margin: 0 auto;
  padding: 10% 5%;
  max-width: 500px;
  background-color: white;
  border: 3px solid #00b8e6;
  border-radius: 8px;
}

.login-reminder {
  font-size: 18px;
}

@media only screen and (min-width: 950px) {
  .LoginPage {
    width: 80%;
  }
}