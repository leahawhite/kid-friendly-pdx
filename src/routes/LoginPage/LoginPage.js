import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import LoginForm from '../../components/LoginForm/LoginForm';
import './LoginPage.css';

export default class LoginPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  }

  handleLoginSuccess = () => {
    const { history } = this.props
    this.props.onLogin()
    history.goBack()
  }

  render() {
    const { onLogin } = this.props
    return (
      <section className="LoginPage">
        <h2>Log In</h2>
        <LoginForm
          onLogin={onLogin}
          onLoginSuccess={this.handleLoginSuccess} />
        <div className="login-reminder">New around here?{' '}
          <Link to='/signup'>
            Sign up.
          </Link>
        </div>
      </section>
    )
  }
}