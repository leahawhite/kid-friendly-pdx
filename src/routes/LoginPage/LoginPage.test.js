import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LoginPage from './LoginPage'

describe('LoginPage component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <LoginPage />
      </BrowserRouter>, 
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders UI as expected', () => {
    const wrapper = shallow(<LoginPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})
