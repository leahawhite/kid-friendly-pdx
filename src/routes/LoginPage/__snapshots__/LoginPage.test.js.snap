// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component renders UI as expected 1`] = `
<section
  className="LoginPage"
>
  <div
    className="LoginPage-container"
  >
    <h2>
      Log In
    </h2>
    <LoginForm
      onLoginSuccess={[Function]}
    />
    <div
      className="login-reminder"
    >
      New around here?
       
      <Link
        to="/signup"
      >
        Sign up.
      </Link>
    </div>
  </div>
</section>
`;
